src/main/scala/it/unibo/authsim/library/dsl/otp/builders/TOTPBuilder.scala

```diff
@@ -0,0 +1,43 @@
+package it.unibo.authsim.library.dsl.otp.builders
+
+import it.unibo.authsim.library.dsl.HashFunction
+import it.unibo.authsim.library.dsl.otp.builders.OTPBuilder.AbstractTOTPBuilder
+import it.unibo.authsim.library.dsl.otp.generator.OTPGenerator
+import it.unibo.authsim.library.dsl.otp.model.{HOTP, TOTP}
+import it.unibo.authsim.library.dsl.otp.util.OTPHelpers.generatorSeed
+import it.unibo.authsim.library.dsl.policy.model.StringPolicies.OTPPolicy
+
+import scala.concurrent.duration.Duration
+
+
+/**
+ * ''TOTPBuilder'' is the Time-based One-Time Password (TOTP) builder.
+ */
+case class TOTPBuilder() extends AbstractTOTPBuilder:
+  override def build: TOTP = new TOTP:
+
+    private var _createDate: Option[Long] = None
+
+    override def length: Int = TOTPBuilder.this._length
+
+    override def createDate: Option[Long] = this._createDate
+
+    override def timeout: Duration = TOTPBuilder.this._timeout
+
+    override def hashFunction: HashFunction = TOTPBuilder.this._hashFunction
+
+    override def policy: OTPPolicy = TOTPBuilder.this._policy
+
+    override def generate: String =
+      val generated = OTPGenerator(this.hashFunction, TOTPBuilder.this._secret, this.length, TOTPBuilder.this._seed)
+      this._createDate = Some(System.currentTimeMillis())
+      generated
```

Comment:
Is there a specific reason to have the variable `generated`?

If not, I suggest this change:
```suggestion
      this._createDate = Some(System.currentTimeMillis())
      OTPGenerator(this.hashFunction, TOTPBuilder.this._secret, this.length, TOTPBuilder.this._seed)
```